packages/@aws-cdk/aws-lambda/lib/layers.ts

```diff
@@ -3,21 +3,10 @@ import { Code } from './code';
 import { CfnLayerVersion, CfnLayerVersionPermission } from './lambda.generated';
 import { Runtime } from './runtime';
 
-export interface LayerVersionProps {
-  /**
-   * The runtimes compatible with this Layer.
-   *
-   * @default - All runtimes are supported.
-   */
-  readonly compatibleRuntimes?: Runtime[];
-
-  /**
-   * The content of this Layer.
-   *
-   * Using `Code.fromInline` is not supported.
-   */
-  readonly code: Code;
-
+/**
+ * Non runtime options
+ */
+export interface LayerVersionOptions {
```

Comment:
Leaving these outstanding questions here for feedback:

1. Should this be LayerVersionBaseProps and can it also include compatibleRuntimes?

Reply:
> 1. Should this be `LayerVersionBaseProps`?

I modelled this interface after lambda's `FunctionOptions` and chose the name in favour of consistent naming.

> 2. Can it also include `compatibleRuntimes`?

I'm on the fence about this one. Adding this member to the interface might impede a construct based on `lambda.LayerVersion` to auto-detect compatible runtimes.

Reply:
> I modelled this interface after lambda's `FunctionOptions` and chose the name in favour of consistent naming.

> I'm on the fence about this one. Adding this member to the interface could impede a construct from auto-detecting compatible runtimes, but I'm not sure that's a problem for this PR.

Fair points. What about this:

```typescript
export interface LayerVersionBaseProps {
  /**
   * The content of this Layer.
   *
   * Using `Code.fromInline` is not supported.
   */
  readonly code: Code;
}

export interface LayerVersionOptions extends LayerVersionBaseProps {
  /**
   * The runtimes compatible with this Layer.
   *
   * @default - All runtimes are supported.
   */
  readonly compatibleRuntimes?: Runtime[];
}
```
> If there should be a different model in play, I'm game. But before going down this rabbit hole, I want to collect more feedback.

Agreed.